comm.go

```diff
@@ -44,6 +46,9 @@ func (p *PubSub) handleNewStream(s inet.Stream) {
 			return
 		}
 
+		measure := metrics.MIncomingMsgs.M((int64)(rpc.Size()))
+		go stats.Record(context.Background(), measure)
```

Comment:
I would recommend against doing this with the go keyword. It removes profiling and tracking context from the runtime, and if you do get performance issues in the Record, you have no backpressure on the number of goroutines. There's also the idea of sequencing. At this point in the code, you want that measure updated, not at some undetermined point in the future. It might be a minor point here in the code, but it sets a precedent that's difficult to back out of.